tests/lib/eslint/eslint.js

```diff
@@ -2917,6 +2917,51 @@ describe("ESLint", () => {
                 assert.strictEqual(results[0].messages[0].ruleId, "post-processed");
             });
 
+            it("should run processors when calling lintText with processor resolves same extension but different content correctly", async () => {
+                let count = 0;
+
+                eslint = new ESLint({
+                    useEslintrc: false,
+                    overrideConfig: {
+                        plugins: ["test-processor"],
+                        rules: {
+                            "no-console": 2,
+                            "no-unused-vars": 2
+                        }
```

Comment:
```suggestion
                        overrides: [{
                            files: ["**/*.txt/*.txt"],
                            rules: {
                                "no-console": 2,
                                "no-unused-vars": 2
                            }
                        }]
```

I think this would confirm that the original issue https://github.com/eslint/eslint/issues/14207 is fixed.